utils/managed_bytes.hh

```diff
@@ -30,215 +30,213 @@
 #include <seastar/util/alloc_failure_injector.hh>
 #include <unordered_map>
 #include <type_traits>
+#include "utils/bit_cast.hh"
 
 template <mutable_view is_mutable_view>
 class managed_bytes_basic_view;
 using managed_bytes_view = managed_bytes_basic_view<mutable_view::no>;
 using managed_bytes_mutable_view = managed_bytes_basic_view<mutable_view::yes>;
 
+// The layout of blob_storage::storage is as follows:
+// If this is the last blob_storage in the list, there is only data.
+// If this is not the last blob_storage in the list, the first bytes of `storage`
+// contain a ref_type pointing to the next fragment, and all the following bytes
+// are data.
+// A blob_storage can learn whether it's the last by looking at the least significant
+// bit of the size field in *backref.
 struct blob_storage {
-    struct [[gnu::packed]] ref_type {
-        blob_storage* ptr;
-
-        ref_type() {}
-        ref_type(blob_storage* ptr) : ptr(ptr) {}
-        operator blob_storage*() const { return ptr; }
-        blob_storage* operator->() const { return ptr; }
-        blob_storage& operator*() const { return *ptr; }
-    };
     using size_type = uint32_t;
     using char_type = bytes_view::value_type;
 
+    struct [[gnu::packed]] ref_type {
+        blob_storage* ptr = nullptr;
+        size_type x = 0; // The least significant bit marks whether the fragment pointed to by ptr is not last. (1 = not last, 0 = last)
```

Comment:
I think the field deserves a better name :)

Reply:
The whole thing is private to managed_bytes, I think it's enough.